src/foundry/common/documents/table-result.mjs.d.ts

```diff
@@ -0,0 +1,153 @@
+import type { documents } from "../module.mjs";
+
+declare global {
+  type TableResultData = BaseTableResult.Properties;
+}
+
+/**
+ * The Document definition for a TableResult.
+ * Defines the DataSchema and common behaviors for a TableResult which are shared between both client and server.
+ */
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface BaseTableResult extends BaseTableResult.Properties {}
+declare class BaseTableResult extends Document<BaseTableResult.SchemaField, BaseTableResult.Metadata> {
+  /**
+   * @param data    - Initial data from which to construct the Table Result
+   * @param context - Construction context options
+   */
+  constructor(data: BaseTableResult.ConstructorData, context?: DocumentConstructionContext);
+
+  static override metadata: Readonly<BaseTableResult.Metadata>;
+
+  static override defineSchema(): BaseTableResult.Schema;
+
+  override testUserPermission(
+    user: foundry.documents.BaseUser,
+    permission: keyof typeof CONST.DOCUMENT_OWNERSHIP_LEVELS | CONST.DOCUMENT_OWNERSHIP_LEVELS,
+    {
+      exact
+    }?: {
+      /**
+       * Require the exact permission level requested?
+       * @defaultValue `false`
+       */
+      exact?: boolean;
+    }
+  ): boolean;
+
+  static override migrateData(source: object): object;
+
+  static override shimData(
+    data: object,
+    {
+      embedded
+    }?: {
+      /**
+       * Apply shims to embedded models?
+       * @defaultValue `true`
+       */
+      embedded?: boolean;
+    }
+  ): object;
+}
+export default BaseTableResult;
+
+declare namespace BaseTableResult {
+  type Metadata = Merge<
+    DocumentMetadata,
+    {
+      name: "TableResult";
+      collection: "results";
+      label: "DOCUMENT.TableResult";
+      labelPlural: "DOCUMENT.TableResults";
+      coreTypes: ["0", "1", "2"];
+      permissions: {
+        update: (user: documents.BaseUser, doc: Document.Any, data: UpdateData) => boolean;
+      };
+    }
+  >;
+
+  type SchemaField = fields.SchemaField<Schema>;
+  type ConstructorData = UpdateData;
+  type UpdateData = fields.SchemaField.InnerAssignmentType<Schema>;
+  type Properties = fields.SchemaField.InnerInitializedType<Schema>;
+  type Source = fields.SchemaField.InnerPersistedType<Schema>;
+
+  interface Schema extends DataSchema {
+    /**
+     * The _id which uniquely identifies this TableResult embedded document
+     * @defaultValue `null`
+     */
+    _id: fields.DocumentIdField;
+
+    /**
+     * A result subtype from CONST.TABLE_RESULT_TYPES
+     * @defaultValue `CONST.TABLE_RESULT_TYPES.TEXT`
+     */
+    type: fields.NumberField<{
+      required: true;
+      choices: CONST.TABLE_RESULT_TYPES[];
+      initial: typeof CONST.TABLE_RESULT_TYPES.TEXT;
+      validationError: "must be a value in CONST.TABLE_RESULT_TYPES";
+    }>;
+
+    /**
+     * The text which describes the table result
+     * @defaultValue see {@link fields.HTMLField}
```

Comment:
This can stay as is, but if I remember right the default is trivially an empty string.

Reply:
I've gone ahead and simplified it for DX